### PR TITLE
Fix beforerequest when is undefined

### PR DESCRIPTION
**Pull request by emilkoto**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -47,26 +47,49 @@ EventResource.prototype.handle = function (ctx, next) {
...
-      ctx.done(err, result);
-    });
-  } else if (ctx.method === "DELETE" && this.events.delete) {
-    this.events.delete.run(ctx, domain, function(err) {
-      ctx.done(err, result);
-    });
-  } else if (ctx.method === "PUT" && this.events.put) {
-    this.events.put.run(ctx, domain, function(err) {
-      ctx.done(err, result);
+  if (this.events.beforerequest !== undefined) {
```
Comment (andreialecu) [inline index.js]: You should just check for `if (this.events.beforerequest) {`

see line 54, 58, 62, etc.

**Comment by andreialecu** [comment]
Also same problem as in your previous PR:

> I think you're missing the beforerequest event here: https://github.com/deployd/dpd-event/pull/20/files#diff-168726dbe96b3ce427e7fedce31bb0bcL11
> 
> Did you test this?

**Comment by emilkoto** [comment]
Sorry,
it had happened to me.

I already made a commit.
Do I do the pullrequest again?

File: index.js [hunk 2]
```diff
@@ -47,26 +47,49 @@ EventResource.prototype.handle = function (ctx, next) {
...
-    this.events.delete.run(ctx, domain, function(err) {
-      ctx.done(err, result);
-    });
-  } else if (ctx.method === "PUT" && this.events.put) {
-    this.events.put.run(ctx, domain, function(err) {
-      ctx.done(err, result);
+  if (this.events.beforerequest) {
+    this.events.beforerequest.run(ctx, domain, (err) => {
+      if (err)
+        ctx.done(err, result);
```
Comment (andreialecu) [inline index.js]: this should have return before it, also result is undefined at this point, pass null

File: index.js [hunk 3]
```diff
@@ -47,26 +47,49 @@ EventResource.prototype.handle = function (ctx, next) {
...
+      if (ctx.method === "POST" && this.events.post) {
```
Comment (andreialecu) [inline index.js]: this code (line 54-72) is duplicated below, extract it to a method inside the function and call the method instead in both places, in order to reuse it.

**Comment by emilkoto** [comment]
I have changed the code, to implement with method.

**Pull request closed without merging**